setup.py

```diff
@@ -43,15 +45,27 @@ class Clean(clean):
 
     def run(self):
         super().run()
-        call(('rm -rf MANIFEST BUILD BUILDROOT SPECS RPMS SRPMS SOURCES',
-              'PYPI_UPLOAD ./build ./dist'), shell=True)
-        call('rm -rf ./man/avocado.1 ./docs/build', shell=True)
-        call('rm -rf /var/tmp/avocado* /tmp/avocado/*', shell=True)
-        call('find . -name "*.egg-info" -exec rm -rv {} +', shell=True)
-        call('find . -name "*.pyc" -exec rm -rv {} +', shell=True)
-        call('find . -name __pycache__ -type d -exec rm -rv {} +', shell=True)
-        call('find ./docs/source/api/ -name "*.rst" -exec rm -rv {} +',
-             shell=True)
+        cleaning_list = ["MANIFEST", "BUILD", "BUILDROOT", "SPECS",
+                         "RPMS", "SRPMS", "SOURCES", "PYPI_UPLOAD",
+                         "./build", "./dist",
+                         "./man/avocado.1", "./docs/build",
+                         "/tmp/avocado/"]
+
+        cleaning_list += list(Path('/var/tmp/').glob(".avocado-task*"))
+        cleaning_list += list(Path('/var/tmp/').glob("avocado*"))
+        cleaning_list += list(Path('.').rglob("*.egg-info"))
+        cleaning_list += list(Path('.').rglob("*.pyc"))
+        cleaning_list += list(Path('.').rglob("__pycache__"))
+        cleaning_list += list(Path('./docs/source/api/').rglob("*.rst"))
+
+        for e in cleaning_list:
+            if os.path.exists(e) and os.path.isfile(e):
+                os.remove(e)
+            if os.path.exists(e) and os.path.isdir(e):
+                remove_tree(e)
```

Comment:
Any reason for using remove_tree instead of `os.removedirs()`? I'm asking this because we are trying to remove distutils, since as pointed on the docs: 

> ... most Python users will not want to use this module directly, but instead use the cross-version tools maintained by the Python Packaging Authority. In particular, setuptools is an enhanced alternative to distutils ...

Reply:
@beraldoleal  `os.removedirs()` doesn't work if the directory is not empty. However, I can use instead `shutil.rmtree`